pdns/recursordist/test-syncres_cc2.cc

```diff
@@ -62,6 +63,22 @@ BOOST_AUTO_TEST_CASE(test_referral_depth)
     BOOST_CHECK(false);
   }
   catch (const ImmediateServFailException& e) {
+    BOOST_CHECK(e.reason.find("max-recursion-depth") != string::npos);
+  }
+
+  // Then check if the setup with high limit is OK.
+  SyncRes::s_maxdepth = 50;
+  try {
+    vector<DNSRecord> ret;
+    int rcode = sr->beginResolve(target, QType(QType::A), QClass::IN, ret);
+    BOOST_CHECK_EQUAL(rcode, RCode::NoError);
+    BOOST_CHECK_EQUAL(ret.size(), 1U);
```

Comment:
This should be a `require` instead of a `check` so we don't read out-of-bounds just after if the test fails:
```suggestion
    BOOST_REQUIRE_EQUAL(ret.size(), 1U);
```